.circleci/Dockerfile

```diff
@@ -1,4 +1,4 @@
-FROM    circleci/golang:1.9
+FROM    circleci/golang:1.10rc1
```

Comment:
that's only ok because none of the artefact produced by circle are used for real deployments
(we shouldn't use pre release software for prod - but it's good to know soon if there are compatibility issues which I am guessing is your goal here)

Reply:
By the time this PR will be reviewed - pretty sure 1.10 will be released :-)
AFAIK it's any time now - so 0.6 can probably use it. 

I'm not sure about what will be used for real deployment - I would expect many advanced users
to build their own (as google does), few download pre-compiled binaries from the net to run
in production. And orgs using CircleCI are likely to just expand what we have.